contracts/src/libraries/ownable.cairo

```diff
@@ -1,110 +1,155 @@
 use starknet::ContractAddress;
 
+// TODO: consider replacing with OwnableTwoStep if https://github.com/OpenZeppelin/cairo-contracts/pull/809/ lands
+
 #[starknet::interface]
-trait IOwnable<TContractState> {
-    fn owner(self: @TContractState) -> ContractAddress;
-    fn proposed_owner(self: @TContractState) -> ContractAddress;
-    fn transfer_ownership(ref self: TContractState, new_owner: ContractAddress);
-    fn accept_ownership(ref self: TContractState);
-    fn renounce_ownership(ref self: TContractState);
+trait IOwnable<TState> {
+    fn owner(self: @TState) -> ContractAddress;
+    fn proposed_owner(self: @TState) -> ContractAddress;
+    fn transfer_ownership(ref self: TState, new_owner: ContractAddress);
+    fn accept_ownership(ref self: TState);
+    fn renounce_ownership(ref self: TState);
 }
 
-#[starknet::contract]
-mod Ownable {
+#[starknet::component]
+mod OwnableComponent {
     use starknet::ContractAddress;
-    use starknet::contract_address_const;
-    use zeroable::Zeroable;
+    use starknet::get_caller_address;
 
     #[storage]
     struct Storage {
-        _owner: ContractAddress,
-        _proposed_owner: ContractAddress,
+        Ownable_owner: ContractAddress,
```

Comment:
how does one access the component storage variable from the contract? if it's through `self.ownable.<storage_var>` I think we can do without the Camelcase prefix

Reply:
I copied this from OZ and figured I'd keep the naming so it'll be a smaller diff if we choose to migrate to their helper later. I can revert if needed